pkg/northbound/gnmi/set.go

```diff
@@ -208,10 +221,41 @@ func (s *Server) buildUpdateResults(targetUpdates mapTargetUpdates,
 		}
 		networkChanges[store.ConfigName(configName)] = changeID
 	}
-
 	return updateResults, networkChanges, nil
 }
 
+func listenForDeviceResponse(changes mapNetworkChanges, target string, name store.ConfigName) (bool, error) {
+	mgr := manager.GetManager()
+	respChan, ok := mgr.Dispatcher.GetResponseListener(topocache.ID(target))
+	if !ok {
+		log.Infof("Device %s not properly registered, not waiting for southbound confirmation", target)
+		return true, nil
+	}
+	response := <-respChan
+	switch eventType := response.EventType(); eventType {
+	case events.EventTypeAchievedSetConfig:
+		return true, nil
+	case events.EventTypeErrorSetConfig:
+		//Removing previously applied Previously applied
+		for t := range changes {
```

Comment:
you should exclude the **current** config (target) from this, and you have to think that some of them might not have been applied yet at this stage

Reply:
that changes has only the successfully passed targets so we don't have to exclude the one that failed because it's not there.

Reply:
OK